src/utilities/bcl/LocalBCL.cpp

```diff
@@ -167,17 +169,29 @@ namespace openstudio{
 
   void LocalBCL::close()
   {
+    // Close the connection to the database if needed.
+    // TODO: something is bugging me here. why are we using a std::shared_ptr and not unique_ptr?
+    // shared_ptr -> reset() should destroy the object (which calls closeConnection) if it's only reference left, which it should...
+    // TODO: probabably not needed
+    instanceInternal()->closeConnection();
+    // reset pointer
     instanceInternal().reset();
   }
 
-  LocalBCL::~LocalBCL()
+  bool LocalBCL::closeConnection()
   {
     // Close the connection to the database if needed
     if (m_connectionOpen) {
       sqlite3_close(m_db);
       m_connectionOpen = false;
     }
+    return true;
+  }
 
+  LocalBCL::~LocalBCL()
+  {
+    // Close the connection to the database if needed
+    closeConnection();
```

Comment:
Didn't change the Dtor, just wrapped a new method closeConnection() that does the same thing